ConsoleProject/src/services/TextureConverter.cs

```diff
@@ -75,8 +75,39 @@ public class TextureConverter {
 
             return texture;
         }
-
+        
         private Bitmap GetBitmapFromPalette(Texture texture) {
+            Bitmap bmp = new Bitmap(texture.Width, texture.Height);
+            int row = 0;
+            int bitsPerPixelMultiplier = 8 / texture.BitsPerPixel;
+
+            List<System.Drawing.Color> colors = new List<System.Drawing.Color>();
+            
+            for (int c = 0; c < texture.Palette.Length / 4; c++) {
+                Color col = Color.FromArgb(texture.Palette[c * 4 + 3], texture.Palette[c * 4], texture.Palette[c * 4 + 1], texture.Palette[c * 4 + 2]);
+                texture.Colors.Add(col);
+            }
+
+            for (int x = 0; x < texture.Unswizzled.Length; x++) {
```

Comment:
Move this to a private method., with return type of Bitmap.
Pass the color array, the unswizzled array and the with as parameter.

Reply:
Isn't this what the method GetBitmapFromPalette is already doing?
Also, not sure I committed this already, but there are now 2 methods: GetIndexedBitmapFromPalette and GetBitmapFromPalette. Each need a different way of iterating through Unswizzled. So, I would rather keep those two methods as they are.